source/SyntheticTopology/SierpinskiObject.lagda

```diff
@@ -0,0 +1,155 @@
+---
+title:          Definition of Sierpinski object synthetic topology
+authors:        ["Ayberk Tosun", "Martin Trucchi"]
+date-started:   2024-05-02
+date-completed: 2024-05-31
+dates-updated:  [2024-05-28]
+---
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import MLTT.Spartan
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Sets
+open import UF.Sets-Properties
+open import UF.Subsingletons
+open import UF.SubtypeClassifier
+
+module SyntheticTopology.SierpinskiObject
+        (fe : Fun-Ext)
+        (pe : Prop-Ext)
+        (pt : propositional-truncations-exist) where
+
+open import Dominance.Definition
+open import UF.Classifiers
+open import UF.Embeddings
+open import UF.Equiv
+open import UF.ImageAndSurjection pt
+open import UF.Logic
+open import UF.Subsingletons-FunExt
+open import UF.Univalence
+open import UF.UniverseEmbedding
+
+open AllCombinators pt fe
+open PropositionalTruncation pt hiding (_∨_)
+
+\end{code}
+
+What is a Sierpiński object? In Martín Escardó´s unpublished note [_Topology via
+higher-order intuitionistic logic_][1], a Sierpiński object is defined in the
+setting of a topos as a subobject of the subobject classifier. This is also
+given in Definition 2.4 of Davorin Lesnik's thesis, who took this unpublished
+note as a starting point for his PhD thesis.
+
+The purpose of this development is to develop these notions in the context of
+HoTT/UF, where we look at subtypes of the subtype classifier. Because we work
+predicatively, however, the definition of the notion of Sierpiński object is not
+that straightforward in our setting.
+
+In the impredicative setting of topos theory, one works with the topos of 𝓤-sets
+over some universe 𝓤, and the Sierpiński object in that setting would translate
+to:
+
+\begin{code}
+
+Sierpinski-Object₀ : (𝓤 : Universe) → 𝓤 ⁺  ̇
+Sierpinski-Object₀ 𝓤 = Subtype' 𝓤 (Ω 𝓤)
+
+\end{code}
+
+However, many dominances that we are interested in do not fit this definition in
+a predicative setting. For example, the largest possible dominance of `Ω 𝓤` is
+𝓤⁺-set rather than a 𝓤-set. Accordingly, we generalize the universes in the
+notion of Sierpiński object as follows:
+
+\begin{code}
+
+Generalized-Sierpinski-Object : (𝓤 𝓥 : Universe) → (𝓤 ⊔ 𝓥) ⁺  ̇
+Generalized-Sierpinski-Object 𝓤 𝓥 = Ω 𝓤 → Ω 𝓥
+
```

Comment:
In this explanation, we first give the generalized definition and then say that we will mostly be working with a certain special case of it.

Maybe we should do the opposite:

1. First explain the special case that arises in practice.
2. Explain that any proof that works in this special case easily generalizes to the generalized definition.
3. We therefore work with the generalized definition, even though the mathematical meaning of the generalized form is not entirely clear.

I think this would make things a lot more clearer.

Reply:
I think it has been modified since. So I guess it is okay. I added the references as well.